src/mono/mono/utils/mono-dl.c

```diff
@@ -168,6 +171,37 @@ fix_libc_name (const char *name)
 }
 #endif
 
+/**
+ * mono_dl_open_self:
+ * \param error_msg pointer for error message on failure
+ *
+ * Returns a handle to the main program, on android x86 it's not possible to 
+ * call dl_open(null), it returns a null handle, so this function returns RTLD_DEFAULT
+ * handle in this platform.
+ */
+MonoDl*
+mono_dl_open_self (char **error_msg)
+{
+#if defined(ENABLE_NETCORE) && defined(TARGET_ANDROID)
+	MonoDl *module;
+	if (error_msg)
+		*error_msg = NULL;
+	module = (MonoDl *) g_malloc (sizeof (MonoDl));
+	if (!module) {
+		if (error_msg)
+			*error_msg = g_strdup ("Out of memory");
+		return NULL;
+	}
+	mono_refcount_init (module, NULL);
+	module->handle = RTLD_DEFAULT;
```

Comment:
I don't think this will work on 32-bit? I thought the glib commit said `dlopen(NULL)` was fine on 64-bit?

Reply:
It doesn't work on 64-bit, me and @steveisok tested and dlopen(null) returns NULL in 32 or 64 bits.